test/test_v2_dataset_warnings.py

```diff
@@ -0,0 +1,22 @@
+import pytest
+
+
+def test_warns_if_imported_from_datasets():
+    import torchvision
+
+    value = torchvision._WARN_ABOUT_BETA_TRANSFORMS
+
+    setattr(torchvision, "_WARN_ABOUT_BETA_TRANSFORMS", True)
+
+    with pytest.warns(UserWarning, match=torchvision._BETA_TRANSFORMS_WARNING):
+        from torchvision.datasets import wrap_dataset_for_transforms_v2
+
+    setattr(torchvision, "_WARN_ABOUT_BETA_TRANSFORMS", value)
```

Comment:
If you want to do it manually, you will need the context manager into a `try / finally` and reset there. Otherwise, you will pollute global state if the context manager errors out, i.e. no warning is raised.